library/sinks/AiSDK.ts

```diff
@@ -0,0 +1,157 @@
+import { Agent } from "../agent/Agent";
+import type { Hooks } from "../agent/hooks/Hooks";
+import { InterceptorObject, wrapExport } from "../agent/hooks/wrapExport";
+import type { Wrapper } from "../agent/Wrapper";
+import { isPlainObject } from "../helpers/isPlainObject";
+
+type PartialAiResponse = {
+  usage: {
+    completionTokens: number;
+    promptTokens: number;
+    totalTokens: number;
+  };
+  response: {
+    modelId: string;
+  };
+};
+
+export class AiSDK implements Wrapper {
+  private inspectAiCall(agent: Agent, args: unknown, response: unknown) {
+    if (!this.isResult(response)) {
+      return;
+    }
+
+    const provider = this.getProviderFromArgs(args);
+    if (!provider) {
+      return;
+    }
+
+    const modelName = this.getModelName(response);
+
+    const aiStats = agent.getAIStatistics();
+    aiStats.onAICall({
+      provider: provider,
+      model: modelName,
+      inputTokens: response.usage.promptTokens,
+      outputTokens: response.usage.completionTokens,
+    });
+  }
+
+  private isResult(result: unknown): result is PartialAiResponse {
+    if (
+      result &&
+      typeof result === "object" && // It is not a plain object
+      !Array.isArray(result) &&
+      "usage" in result &&
+      isPlainObject(result.usage) &&
+      typeof result.usage.completionTokens === "number" &&
+      typeof result.usage.promptTokens === "number" &&
+      typeof result.usage.totalTokens === "number" &&
+      "response" in result &&
+      result.response &&
+      isPlainObject(result.response) &&
+      typeof result.response.modelId === "string"
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  private getProviderFromArgs(args: unknown): string | undefined {
```

Comment:
```suggestion
  private getProviderFromArgs(args: unknown[]): string | undefined {
```

Removes the need for the Array.isArray check below?